apps/desktop/src/lib/stacks/stacksService.ts

```diff
@@ -0,0 +1,50 @@
+import { reduxApi } from '$lib/redux/api';
+import { DesktopRedux } from '$lib/redux/store.svelte';
+import type { Unsubscribe } from '$lib/redux/unsubscribe';
+import type { Stack } from './stack';
+
+/** Poll every five minutes to mitigate potential bugs. */
+const POLLING_INTERVAL = 5 * 60 * 1000;
+
+enum Tags {
+	Stacks = 'Stacks'
+}
+
+export class StacksService {
+	private stacksApi = reduxApi.injectEndpoints({
+		endpoints: (build) => ({
+			/** Fetches all stacks. */
+			get: build.query<Stack[], { projectId: string }>({
+				query: ({ projectId }) => ({ command: 'stacks', params: { projectId } }),
+				providesTags: [Tags.Stacks]
+			}),
+			new: build.mutation<Stack, { projectId: string }>({
+				query: ({ projectId }) => ({
+					command: 'create_virtual_branch',
+					params: { projectId, branch: {} }
+				}),
+				invalidatesTags: [Tags.Stacks]
+			})
+		})
+	});
+
+	constructor(private state: DesktopRedux) {}
+
+	poll(projectId: string): Unsubscribe {
+		return this.state.dispatch(
+			this.stacksApi.endpoints.get.initiate(
+				{ projectId },
+				{ subscribe: true, subscriptionOptions: { pollingInterval: POLLING_INTERVAL } }
```

Comment:
Why polling? Consider it is probably better to never poll these because we wish the app to be deterministic

Reply:
cool cool